The Microfinance Gateway.js

```diff
@@ -1,109 +1,173 @@
+
+	***** END LICENSE BLOCK *****
 */
 
 
-/** Articles */
-FW.Scraper({
-itemType         : 'report',
-detect           : FW.Xpath('//span[@class="blurb_title_verdana"]'),
-title            : FW.Xpath('//span[@class="blurb_title_verdana"]').text().trim(),
-attachments      : [{ url: FW.Url(),
-  title:  "Microfinance Gateway Snapshot",
-  type: "text/html" },
-  {url: FW.Xpath('//p[@class="blurb_title"]/a[contains(text(), "Download this Document")]/@href').text(), title: "Microfinance Gateway Fulltext"}
-  ],
-creators         : FW.Xpath('//p[span[@class="blurb_title_verdana"]]/text()').text().trimInternal().remove(/.+?,/).remove(/\(.*?\)/).split(/\s*&\s*|\.,/).cleanAuthor("author", true),
-date             : FW.Xpath('//p[span[@class="blurb_title_verdana"]]/text()').text().remove(/,.+/),
-institution      : FW.Xpath('//p[b[contains(text(), "Publisher")]]/a').text(),
-abstractNote	 : FW.Xpath('//p[span[@class="blurb_title_verdana"]]/following-sibling::p').text().trim()
-});
- 
-FW.MultiScraper({
-itemType         : 'multiple',
-detect           : FW.Url().match(/\?type=SearchPortlet/),
-choices          : {
-  titles :  FW.Xpath('//span[@class="blurb_title"]/a').text(),
-  urls    :  FW.Xpath('//span[@class="blurb_title"]/a').key("href")
+function detectWeb(doc, url) {
+	if (url.indexOf('/library/')>-1) {
+		return "report";
+		//other types as events or press releases are ignored
+	} else if (url.indexOf('/site-search?')>-1 && getSearchResults(doc, true)) {
+		return "multiple";
+	}
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = ZU.xpath(doc, '//div[contains(@class, "views-row")]/div[contains(@class, "field-name-title-field")]/a[contains(@href, "/library/")]');
+	for (var i=0; i<rows.length; i++) {
+		var href = rows[i].href;
+		var title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
 }
-});
-	/** BEGIN TEST CASES **/
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+
+function scrape(doc, url) {
+	var item = new Zotero.Item("report");
+	item.title = ZU.xpathText(doc, '//main//h1');
+	item.date = ZU.xpathText(doc, '//span[contains(@class, "date-display-single")]');
+	var authorString = ZU.xpathText(doc, '//div[contains(@class, "field-name-field-publication-author")]/text()');
+	//e.g. Hirschland, M., Jazayeri, A. & Lee, N.
+	if (authorString) {
+		var authors = authorString.split(/\.,|&/);
+		for (var i=0; i<authors.length; i++) {
+			item.creators.push(ZU.cleanAuthor(authors[i], "author"));
```

Comment:
I think you need a `, true` here. note the authors in the tests have lastName & firstName swapped